src/common/models/dimension/dimension.ts

```diff
@@ -89,7 +100,8 @@ export class Dimension implements Instance<DimensionValue, DimensionJS> {
       name: this.name,
       title: this.title,
       expression: this.expression.toJS(),
-      kind: this.kind
+      kind: this.kind,
+      url: this.url
```

Comment:
oh, one more thing if there is something very optional like `url` in an immutable class that defaults to null. it might be better to not put it on the JS.
do `if (this.url) js.url = this.url;`